src/main/models/vehicles/Train.java

```diff
@@ -0,0 +1,48 @@
+package main.models.vehicles;
+
+import main.models.vehicles.interfaces.TrainLoading;
+
+public class Train implements TrainLoading {
+    private final TrainSize trainSize;
+
+    private int currentCapacity;
+    private int profit = 0;
+
+    public Train(TrainSize trainSize){
+        this.trainSize = trainSize;
+        this.currentCapacity = trainSize.getCapacity();
+    }
+
+    @Override
+    public boolean loadVehicle(Vehicle vehicle) {
+        if(this.trainSize.getVehicleSizeType() != vehicle.getVehicleSize()){
+            System.err.println("Vehicle doesn't match train vehicle size type!");
+            return false;
+        }else if(this.currentCapacity == 0){
+            System.err.println("Train is in capacity.\nSorry, your vehicle can't be loaded on this train.");
+            return false;
+        }
+        this.profit += vehicle.getPayAmount();
```

Comment:
Should a train or a terminal have the responsibility to store the profit? Think about this, this is not necessarily wrong. :)

Reply:
For me, It makes sense for train to store the profit but of course it could be segregated in other class.